### DISCUSSION TITLE
Forced 2FA error for first time login

### DISCUSSION THREAD

**Discussion by Programmierschmied** [requarks/wiki#4274] [Error / Bug Report]
If I enable forced 2fa and tell the user to change his password after first login then it results into the error "invalid token".
Did try it with iOS Android and Google Authenticator / FreeOTP

The error doesn't occur if I got forced 2fa enabled and don't let the user change his password after first login.

**Comment by PeachesMLG** [comment]
Getting the same error, if I create a new user, and tick the require change password on login box, It runs into a lot of issues with the 2fa.

If I disable require 2fa
create the user with require change password
the user logs in and changes their password
and re-enable requre 2fa

Then everything works as expected 

I have a video recording of the bug in question here.

https://user-images.githubusercontent.com/26843204/228647705-a273ba69-0a1f-47c5-935c-bdfe128eed72.mp4


Interestingly enough, it looks like the first time I submit the 2fa it succeeds (No error message appears, and my user says that 2fa is enabled successfully) you can also see the forum in the background change to prompt a new password to be inputed, however the 2fa box doesnt close.

Would a simple fix for this be to change the order of events in which this happens? Reset the password first and then enable 2fa. I dont know much about the structure of this code, but it seems like this would be an easy way to prevent this error
